src/CukeCommands.cpp

```diff
@@ -34,11 +38,28 @@ void CukeCommands::endScenario() {
 }
 
 const std::string CukeCommands::snippetText(const std::string stepKeyword, const std::string stepName) const {
-    std::stringstream snippetText; // TODO Escape stepName
-    snippetText << boost::to_upper_copy(stepKeyword) << "(\"^" << stepName << "$\") {" << std::endl;
-    snippetText << "    pending();" << std::endl;
-    snippetText << "}" << std::endl;
-    return snippetText.str();
+    const boost::regex esc("[\".^$|()\\[\\]{}*+?\\\\]");
+    const std::string rep("\\\\&");
+#if __cplusplus <= 199711L
+    std::vector<std::string> pieces;
+    pieces.push_back(boost::to_upper_copy(stepKeyword));
+    pieces.push_back("(\"^");
+    pieces.push_back(regex_replace(stepName, esc, rep, boost::match_default | boost::format_sed));
+    pieces.push_back("$\") {\n"
+                     "    pending();\n"
+                     "}\n"
+                     );
+#else //TODO: check if some unit test are failing because of this
+    std::array<std::string, 6> pieces = {
+        boost::to_upper_copy(stepKeyword),
+        "(\"^",
+        regex_replace(stepName, esc, rep, boost::match_default | boost::format_sed),
+        "$\") {\n",
+        "    pending();\n",
+        "}\n"
+    };
+#endif
+    return boost::algorithm::join(pieces, "");
```

Comment:
It was spontaneous - i've found out that boost has join and wanted to check it out ;) In this place I miss convenience of QStrings, for example method .arg ideal for such  string joining.